File: backend/Ledgerly.Application/DTOs/Payment/PaymentDto.cs
namespace Ledgerly.Application.DTOs.Payment;

public class PaymentDto
{
    public Guid Id { get; set; }
    public Guid OrganizationId { get; set; }
    public Guid BillId { get; set; }
    public Guid CustomerId { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public DateTime PaymentDate { get; set; }
    public string? Reference { get; set; }
    public Guid? ReceiptFileId { get; set; }
    public string? ReceiptFileUrl { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: app/lib/models/payment/payment_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment_dto.g.dart';

@JsonSerializable()
class PaymentDto {
  final String id;
  final String organizationId;
  final String billId;
  final String customerId;
  final String customerName;
  final double amount;
  @JsonKey(name: 'paymentDate')
  final DateTime paymentDate;
  final String? reference;
  final String? receiptFileId;
  final String? receiptFileUrl;
  @JsonKey(name: 'createdAt')
  final DateTime createdAt;

  PaymentDto({
    required this.id,
    required this.organizationId,
    required this.billId,
    required this.customerId,
    required this.customerName,
    required this.amount,
    required this.paymentDate,
    this.reference,
    this.receiptFileId,
    this.receiptFileUrl,
    required this.createdAt,
  });

  factory PaymentDto.fromJson(Map<String, dynamic> json) =>
      _$PaymentDtoFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentDtoToJson(this);
}

